integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py

```diff
@@ -191,6 +258,10 @@ def to_dict(self) -> Dict[str, Any]:
             hosts=self._hosts,
             custom_mapping=self._custom_mapping,
             index=self._index,
+            api_key=self._api_key.to_dict() if self._api_key and not isinstance(self._api_key, TokenSecret) else None,
```

Comment:
I don't think we need this check for `TokenSecret`. We've been following the policy that we expect `to_dict` to fail if someone tries to serialize a component or a doc store and passed in a Secret constructed from a token rather than an env variable.

Reply:
ok, I will change it so it fails on purpose